x-pack/legacy/plugins/infra/public/pages/logs/analysis/page_setup_content.tsx

```diff
@@ -4,13 +4,212 @@
+              />
+            </EuiText>
+            <EuiSpacer size="l" />
+            <EuiDescribedFormGroup
+              idAria="timeRange"
+              title={
+                <FormattedMessage
+                  id="xpack.infra.analysisSetup.timeRangeTitle"
+                  defaultMessage="Choose a time range"
+                />
+              }
+              description={
+                <FormattedMessage
+                  id="xpack.infra.analysisSetup.timeRangeDescription"
+                  defaultMessage="By default, Machine Learning will analyze log messages starting now and continue indefinitely. You can specify an earlier date to begin, a specific point in time to end, or both."
+                />
+              }
+            >
+              <EuiFormRow
+                describedByIds={['timeRange']}
+                error={false}
+                fullWidth
+                isInvalid={false}
+                label={startTimeLabel}
+              >
+                <EuiFlexGroup gutterSize="s">
+                  <EuiDatePicker
+                    showTimeSelect
+                    selected={startTime}
+                    onChange={setStartTime}
+                    placeholder={startTimeDefaultDescription}
+                    maxDate={now}
+                  />
+                  {startTime && (
+                    <EuiButtonIcon
+                      color="danger"
+                      iconType="cross"
+                      aria-label={clearStartTimeLabel}
+                      onClick={() => setStartTime(null)}
+                    />
+                  )}
+                </EuiFlexGroup>
+              </EuiFormRow>
+              <EuiFormRow
+                describedByIds={['timeRange']}
+                error={false}
+                fullWidth
+                isInvalid={false}
+                label={endTimeLabel}
+              >
+                <EuiFlexGroup gutterSize="s">
+                  <EuiDatePicker
+                    showTimeSelect
+                    selected={endTime}
+                    onChange={setEndTime}
+                    placeholder={endTimeDefaultDescription}
+                    openToDate={now}
+                    minDate={now}
+                    minTime={
+                      selectedEndTimeIsToday
+                        ? now
+                        : moment()
+                            .hour(0)
+                            .minutes(0)
+                    }
+                    maxTime={moment()
+                      .hour(23)
+                      .minutes(59)}
+                  />
+                  {endTime && (
+                    <EuiButtonIcon
+                      color="danger"
+                      iconType="cross"
+                      aria-label={clearEndTimeLabel}
+                      onClick={() => setEndTime(null)}
+                    />
```

Comment:
This remove UX is confusing, I think -- it read to me like an error state and I thought I'd done something wrong. EUI has a form input style for this kind of thing that looks like this:

<img width="425" alt="Screen Shot 2019-08-16 at 1 20 55 PM" src="https://user-images.githubusercontent.com/159370/63188010-4af35380-c02e-11e9-9a51-4509e04a5559.png">

but the only way to add that style (among other form input control styles like loading, etc) is to wrap it in `EuiFormControlLayout`, which may not work with the `EuiDatePicker` component. Docs here: https://elastic.github.io/eui/#/forms/form-controls (scroll to bottom, no way to link to the Form Control Layout section).

@elastic/eui-design may have a better suggestion here?

Reply:
FWIW I made a PR to EUI to make this simpler for us. If that lands, we just add an `onClear` prop to the `EuiDatePicker` which will apply the icon and call whatever we want on click of that x icon. 

https://github.com/elastic/eui/pull/2235

Reply:
This works with `EuiDatePicker` actually!